shared-bindings/keypad/KeyMatrix.c

```diff
@@ -82,20 +83,25 @@
 //|           maximum number of key transition events that are saved.
 //|           Must be >= 1.
 //|           If a new event arrives when the queue is full, the oldest event is discarded.
+//|         :param int debounce_threshold: Emit events for state changes only after a key has been
+//|           in the respective state for ``debounce_threshold`` times on average.
+//|           Successive measurements are spaced apart by ``interval`` seconds.
+//|           The default is 1, which resolves immediately. The maximum is 127.
```

Comment:
I think this is clearer; it still takes a bit of head-scratching to understand the algorithm. A diagram would theoretically help, maybe, but let's not go there yet.

The current defaults are `interval=0.020` and `debounce_threshold=1`. What values would you choose on a typical keyboard to make, say, the rhythm game you are talking about work better. Or what are you choosing for your own use now?

Reply:
Ok, rebased on `main` and integrated the demux keypad.

I also spend a couple of hours testing almost every switch and button I had lying around; that includes multiple keyboard switches, through-hole and face-plate-mounted buttons, and a rubber dome pad. Surprisingly, almost all read consistently with `interval=0.001` and `debounce_threshold=3`. The noisiest switches required `debounce_threshold=5`.
From a typing perspective, I can't tell the difference between threshold values 3 and 5 (also the USB polling interval is 8ms, so no surprise there) -- The latter should be a sensible default.
(The rythm game was an example from another user, I can't verify an improvement there. Typing seems ever so slighly more responsive, although I couldn't do a double blind test.)

Reply:
I appreciate your testing. Did you try with a longer interval, such as the 0.008 you suggested? I think I will leave the current defaults alone for now. The overhead of the interrupts is lower with 0.020, and for most people, the current defaults seem fine. I did some testing with tactile switches and keyswitches when I chose that value. Those who are using `keypad` for a performance keyboard can decrease the polling interval. It would be good to write this up somewhere once it's merged.

Reply:
It would be a fine topic for https://adafruit-playground.com/, and we could point to it from the keypad Learn Guide.